### PR TITLE
Move QSP/GQSP helper code into `qualtran.linalg`

### PR DESCRIPTION
**Pull request by anurudhp**
Moved helper functions that apply more generally to polynomials. As we now have two ways (and more incoming) to compute QSP complementary polynomials, it's better to keep the generic functions separate from the specific bloqs.

### CODE REVIEW

**Review by tanujkhattar — APPROVED** [review]

File: qualtran/linalg/polynomial/qsp_util.py
Comment (mpharrigan) [inline qualtran/linalg/polynomial/qsp_util.py:1]: I never like "util" in a module name. `linalg/polynomial/qsp.py`? or `qsp_polynomial`? Or factor out all the testing functions into `linalg.polynomial.qsp_testing`

File: qualtran/linalg/testing.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import numpy as np
+from numpy.typing import NDArray
+
+
+def assert_matrices_almost_equal(A: NDArray, B: NDArray, *, atol: float = 1e-5):
```
Comment (mpharrigan) [inline qualtran/linalg/testing.py:18]: docstring? worth mentioning that this is based on the norm which is unlike e.g. `np.testing.assert_almost_equal`

Comment (anurudhp) [reply qualtran/linalg/testing.py:18]: added docstring

File: qualtran/linalg/polynomial/qsp_util.py
Comment (anurudhp) [reply qualtran/linalg/polynomial/qsp_util.py:1]: renamed to `qsp_testing`, and moved one non-testing function out to `linalg/polynomial/basic.py`

**Pull request merged**